gips/0081-indexing-payments.md

```diff
@@ -0,0 +1,165 @@
+
+## Prior Art
+
+GIP-0058 presents a previous version of a very similar proposal. This GIP re-introduces the same concepts, but incorporates the latest approach in Graph Horizon, and simplifies the proposal with an off-chain MVP while clarifying that the actor engaging in **Indexing Payments** is generally a Gateway running automated indexer selection.
+
+## High-Level Description
+
+We are proposing a new coordination mechanism to incentivize subgraph serving to a performant level. Such a coordination mechanism is necessary to jump-start, and sometimes maintain, performant subgraph serving. One such time is when a subgraph is being upgraded. In early conversations this has been referred to as **Direct Indexing Payments** (DIPs).
+
+In particular, the coordination mechanism has been designed with the following in mind:
+
+- The objective is to get subgraphs served through the protocol to a sufficiently performant level
+- The price to get a subgraph served at said level should ideally be related to the amount of work required.
+
+While the proposal is meant to fully support this use case, it does not exclude other use cases that may organically arise. Several components are general and can be re-used.
+
+Looking further into the above, we assert that as described, it has some concerns that we need to address:
+
+1. Desired performance should be defined by the payer.
+1. Performance can only be assessed after the service has been consumed.
+1. The cost of serving a subgraph is unknown a-priori, and it depends on the desired query performance as well as the indexing cost.
+
+To solve for the first two, we propose that the gateway is the one coordinating with the indexers. Gateways can define and assess Quality of Service (QoS) on an ongoing basis and decide which indexers to continue to work with based on that.
+
+To solve for the last concern, we propose that indexers are paid proportionally to the amount of work they do to index the subgraph. This amount of work is reported by indexers and thus needs to be sufficiently verifiable for gateways, fishermen or other indexers to dispute how much work was actually done.
+
+In particular we propose that the coordination mechanism is underpinned by an **Indexing Agreement** to be drafted by the gateway, shared with a subset of indexers of its choosing and subsequently approved by some of those indexers. These agreements are designed to allow for indexers to work and collect for their work without having to go back to the payer. They are self-contained in that regard. Furthermore, they specify the following:
```

Comment:
I think here 'without having to go back to the payer' might be stated as 'without needing to trust the payer', or alternative with a trust-minimised dependency for payment?

Reply:
done